packages/itwin/ec3-widget/src/components/ExportModal.tsx

```diff
@@ -0,0 +1,206 @@
+/*---------------------------------------------------------------------------------------------
+* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+* See LICENSE.md in the project root for license terms and full copyright notice.
+*--------------------------------------------------------------------------------------------*/
+import "./ExportModal.scss";
+import React, {
+  useCallback,
+  useEffect,
+  useRef,
+  useState,
+} from "react";
+import { IModelApp } from "@itwin/core-frontend";
+import {
+  Button,
+  Modal,
+  ProgressLinear,
+  ProgressRadial,
+  Text,
+  toaster,
+} from "@itwin/itwinui-react";
+import { useEC3JobsClient } from "./api/context/EC3JobsClientContext";
+import { Link, EC3JobCreate, EC3Job, CarbonUploadState } from "@itwin/insights-client";
+
+interface ExportProps {
+  projectName: string;
+  isOpen: boolean;
+  close: () => void;
+  templateId: string | undefined;
+  token: string | undefined;
+}
+
+const ExportModal = (props: ExportProps) => {
+  const PIN_INTERVAL = 1000;
+  const ec3JobsClient = useEC3JobsClient();
+
+  const [jobStatus, setJobStatus] = useState<CarbonUploadState>();
+  const [jobLink, setJobLink] = useState<Link>();
+
+  const intervalRef = useRef<number>();
```

Comment:
I don't think this interval is cleared if user would manage to close this component before the job finishes.

Reply:
Applied the suggested ```UseEffect``` sollution